engine/texc/src/texclibrary_jni.cpp

```diff
@@ -0,0 +1,329 @@
+        {
+            obj = C2J_CreateBuffer(env, types, buffer);
+            dmTexc::DestroyBuffer(buffer);
+        }
+
+        //env->ReleaseByteArrayElements(array, array_data, JNI_ABORT);
+    DM_JNI_GUARD_SCOPE_END(return 0;);
+    return obj;
+}
+
+// *************************************************************************************************************
+
+// JNIEXPORT jobject JNICALL Java_TexcLibraryJni_CreateImage(JNIEnv* env, jclass cls, jstring path, jbyteArray bytes, jint width, jint height, jint depth, jint numChannels)
+// {
+//     dmLogDebug("%s: env = %p\n", __FUNCTION__, env);
+//     //DM_SCOPED_SIGNAL_CONTEXT(env, return 0;);
+
+//     jobject jimage;
+//     DM_JNI_GUARD_SCOPE_BEGIN();
+
+//         dmTexc::jni::ScopedContext jni_scope(env);
+//         dmTexc::jni::TypeInfos* types = &jni_scope.m_TypeInfos;
+//         //jimage = LoadFromBufferInternal(env, cls, _path, array, data_resolver);
+//         jimage = dmTexc::jni::C2J_CreateImage(env, types, path, bytes, width, height, depth, numChannels);
+//     DM_JNI_GUARD_SCOPE_END(return 0;);
+//     return jimage;
+// }
+
+
+JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void* reserved)
+{
+    dmLogDebug("JNI_OnLoad ->\n");
+    //dmJNI::EnableDefaultSignalHandlers(vm);
+
+    JNIEnv* env;
+    if (vm->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_8) != JNI_OK) {
+        printf("JNI_OnLoad GetEnv error\n");
+        return JNI_ERR;
+    }
+
+    // Find your class. JNI_OnLoad is called from the correct class loader context for this to work.
+    jclass c = env->FindClass( JAVA_PACKAGE_NAME "/TexcLibraryJni");
+    dmLogDebug("JNI_OnLoad: c = %p\n", c);
+    if (c == 0)
+        return JNI_ERR;
+
+    #define JNIFUNC(NAME, SIGNATURE) {(char*) #NAME, (char*)(SIGNATURE), reinterpret_cast<void*>(Java_TexcLibraryJni_ ## NAME )}
+
+    // Register your class' native methods.
+    // Don't forget to add them to the corresponding java file (e.g. ModelImporter.java)
+    static const JNINativeMethod methods[] = {
+        // OLD API
+        JNIFUNC(CreateTexture,          "(Ljava/lang/String;IIIII[B)J"),
+        JNIFUNC(DestroyTexture,         "(J)V"),
+        JNIFUNC(GetHeader,              "(J)L" CLASS_NAME "$Header;"),
+        JNIFUNC(GetDataSizeCompressed,  "(JI)I"),
+        JNIFUNC(GetDataSizeUncompressed,"(JI)I"),
+        JNIFUNC(GetData,                "(J)[B"),
+        JNIFUNC(GetCompressionFlags,    "(J)I"),
+
+
+        JNIFUNC(Resize,                 "(JII)Z"),
+        JNIFUNC(PreMultiplyAlpha,       "(J)Z"),
+        JNIFUNC(GenMipMaps,             "(J)Z"),
+        JNIFUNC(Flip,                   "(JI)Z"),
+
+        JNIFUNC(Encode,                 "(JIIIIZI)Z"),
+
+        // Font glyph buffers
+        JNIFUNC(CompressBuffer,         "([B)L" CLASS_NAME "$Buffer;"),
+
+
+        // Image api
+        //{(char*)"CreateImage", (char*)"(Ljava/lang/String;[BIIII)L" CLASS_NAME "$Scene;", reinterpret_cast<void*>(Java_TexcLibraryJni_CreateImage)},
+    };
```

Comment:
Is this the "new" api?

Reply:
That was some wip idea for the next PR, will remove from this PR.
See your branch for the new api.